lib/charms/acme_client_operator/v0/acme_client.py

```diff
@@ -0,0 +1,147 @@
+```shell
+cd some-charm
+charmcraft fetch-lib charms.acme_client_operator.v0.acme_client
+```
+Then, to initialise the library:
+```python
+from charms.acme_client_operator.v0.acme_client import AcmeClient
+from charms.observability_libs.v0.kubernetes_service_patch import KubernetesServicePatch
+from ops.charm import CharmBase
+from ops.main import main
+from charms.tls_certificates_interface.v1.tls_certificates import (  # type: ignore[import]
+    CertificateCreationRequestEvent,
+    TLSCertificatesProvidesV1,
+)
+class ExampleAcmeCharm(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.tls_certificates = TLSCertificatesProvidesV1(self, "certificates")
+        self.framework.observe(
+            self.tls_certificates.on.certificate_creation_request,
+            self._acme_client_operator.on_certificate_creation_request,
+        )
+        lego_cmd = f"lego --email example@email.com --dns namecheap --domains example.com"
+        self._acme_client_operator = AcmeClient(self, lego_cmd, self.additional_config)
+```
+Charms that leverage this library also need to specify a `provides` relation in their
+`metadata.yaml` file. For example:
+```yaml
+provides:
+  certificates:
+    interface: tls-certificates
+```
+"""
+
+# The unique Charmhub library identifier, never change it
+LIBID = "b3c9913b68dc42b89dfd0e77ac57236d"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+
+import logging
+from typing import Dict
+
+from charms.tls_certificates_interface.v1.tls_certificates import (  # type: ignore[import]
+    CertificateCreationRequestEvent,
+)
+from cryptography import x509
+from cryptography.x509.oid import NameOID
+from ops.charm import CharmBase
+from ops.framework import Object
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+from ops.pebble import ExecError
+
+logger = logging.getLogger(__name__)
+
+LEGO_CERTS_PATH = "/tmp/.lego/certificates/"
+class AcmeClient(Object):
+    """TODO"""
+
+    def __init__(
+        self,
+        charm: CharmBase,
+        cmd: str,
+        csr_path: str,
+        additional_config: Dict[str, str] = {},
+    ):
+        super().__init__(charm, None)
+        self._charm = charm
+        self._csr_path = csr_path
+        self._container_name = self.service_name = self._charm.meta.name
+        self._container = self._charm.unit.get_container(self._container_name)
+        service_name_with_underscores = self.service_name.replace("-", "_")
+        self._cmd = cmd
+        pebble_ready_event = getattr(
+            self._charm.on, f"{service_name_with_underscores}_pebble_ready"
+        )
+        self.framework.observe(pebble_ready_event, self._on_acme_client_pebble_ready)
+        self._additional_config = additional_config
```

Comment:
I prefer the original naming `plugin_config`, as this will only be used by the chosen plugin.

Reply:
I agree

Reply:
https://github.com/ghislainbourgeois/lego_operator/pull/4 
Fixed in the new PR